presto-orc/src/test/java/com/facebook/presto/orc/AbstractTestOrcReader.java

```diff
@@ -211,9 +213,15 @@ public void testCaching()
                 .expireAfterAccess(new Duration(10, MINUTES).toMillis(), MILLISECONDS)
                 .recordStats()
                 .build();
-        StripeMetadataSource stripeMetadataSource = new CachingStripeMetadataSource(new StorageStripeMetadataSource(), stripeFootercache, stripeStreamCache);
+        Cache<StripeReader.StripeStreamId, List<RowGroupIndex>> rowGroupIndexCache = CacheBuilder.newBuilder()
+                .maximumWeight(new DataSize(1, MEGABYTE).toBytes())
+                .weigher((id, rowGroupIndices) -> toIntExact(((List<RowGroupIndex>) rowGroupIndices).stream().mapToLong(RowGroupIndex::getRetainedSizeInBytes).sum()))
+                .expireAfterAccess(new Duration(10, MINUTES).toMillis(), MILLISECONDS)
+                .recordStats()
+                .build();
+        StripeMetadataSource stripeMetadataSource = new CachingStripeMetadataSource(new StorageStripeMetadataSource(), stripeFootercache, stripeStreamCache, rowGroupIndexCache);
 
-        try (TempFile tempFile = createTempFile()) {
+        try (TempFile tempFile = createTempFile(20000)) {
```

Comment:
Does this actual number of records matter in the test? Could we reduce it? I am wondering if writing 20000 record is too expensive in a unit test.

Reply:
changed to 10001. The code path is only triggered when a stripe has more than 1 row group, and each row group has 10K rows. I didn't find a easy way to change the size of the row group.